_posts/2020-03-11-how-to-avoid-null-reference-exception.md

```diff
@@ -0,0 +1,127 @@
+---
+layout: post
+title: "How to avoid NullReferenceException in C#"
+date: 2020-03-11 03:57:07
+tags:
+- .NET
+- C#
+categories:
+- development
+twitter_text: How to avoid NullReferenceException in C#
+authors: Cheng Yang
+image: https://images.unsplash.com/photo-1510915228340-29c85a43dcfe?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1050&q=80
+image_url: https://unsplash.com/photos/9SoCnyQmkzI
+image_credit: Jefferson Santos
+---
+
+I have been working as a software developer for almost three years, the most common exception or bug I made is [NullReferenceException](https://docs.microsoft.com/en-us/dotnet/api/system.nullreferenceexception?view=netframework-4.8) -`System.NullReferenceException: Object reference not set to an instance of an object.` This exception is thrown when you try to access any properties / methods/ indexes on a type of object which points to null. 
+
+## Common Scenario 1:
+
+```csharp
+using System; 
+public class Program
+{
+    public static void Main()
+    {
+     	 string dog = null;
+		   var value = dog.ToString(); //Object reference not set to an instance of an object
+   	   Console.WriteLine(value);
+    }
+}
+```
+In the example above, we try to call the `ToString()` method, it will throw a `NullReferenceException` because `dog` is pointing to null.
+
+## Common Scenario 2:
+```csharp
+using System;
+
+public class Dog
+{
+    public string Breed { get; set; }
+	public int age { get; set; }
+}
+public class Dogs
+{
+    public Dog dog { get; set; }
+}
+
+public class Example
+{
+   public static void Main()
+   {
+      Dogs dog1 = new Dogs();
+      int dogAge = dog1.dog.age; // Object reference not set to an instance of an object
+	   
+      Console.WriteLine(dogAge.ToString());  
+   }
+}
+
+```
+In the example above, you will get `NullReferenceException` because `Dogs` property is null ,there is no way to get the data.
```

Comment:
nit:

```suggestion
In the example above, you will get `NullReferenceException` because `Dogs` property is null, there is no way to get the data.
```